### PR TITLE
v2: make device class concrete

### PR DESCRIPTION
**Pull request by RAYemelyanova**
resolves https://github.com/bluesky/ophyd/issues/1090

made Device a concrete class rather that ABC, simplified logic in corresponding child classes like StandardReadable and DeviceVector.

### CODE REVIEW

**Comment by RAYemelyanova** [comment]
Just rebased it now... going to check why stuff is failing

File: ophyd/v2/core.py [hunk 1]
```diff
@@ -129,15 +129,19 @@ def wrap_f(self) -> AsyncStatus:
...
     #: The parent Device if it exists
     parent: Optional[Device] = None
 
     @property
-    @abstractmethod
     def name(self) -> str:
         """Return the name of the Device"""
+        return self._name
 
     @abstractmethod
```
Comment (coretl) [inline ophyd/v2/core.py:146]:
```suggestion
```

File: ophyd/v2/core.py [hunk 2]
```diff
@@ -148,6 +152,8 @@ def set_name(self, name: str):
         name:
             New name to set
         """
+        self._name = name
+        name_children(self, name)
 
     @abstractmethod
```
Comment (coretl) [inline ophyd/v2/core.py:158]:
```suggestion
```

**Review by coretl — CHANGES_REQUESTED** [review]

File: ophyd/v2/tests/test_core.py [hunk 3]
```diff
@@ -56,53 +56,43 @@ async def test_async_status_success():
...
 
-    @property
-    def name(self):
-        return self._name
-
-    def set_name(self, name: str = ""):
-        self._name = name
-
-    async def connect(self, sim=False):
+    async def connect(self, prefix: str = "", sim=False):
```
Comment (coretl) [inline ophyd/v2/tests/test_core.py:63]:
```suggestion
    async def connect(self, sim=False):
```

**Review by coretl — APPROVED** [review]

**Pull request merged**